### PR TITLE
ISPN-7844 Cache templates with Type fail when they enable indexing

### PR DESCRIPTION
**Pull request by wburns**
* Make sure QueryInterceptor abides by Inject rules

https://issues.jboss.org/browse/ISPN-7844

### CODE REVIEW

**Comment by wburns** [comment]
Letting CI run to make sure I didn't break anything accidentally.

**Comment by anistor** [comment]
Looks good.

File: query/src/main/java/org/infinispan/query/backend/QueryInterceptor.java [hunk 1]
```diff
@@ -112,15 +113,17 @@ protected void injectDependencies(TransactionManager transactionManager,
...
+      this.internalCacheRegistry = internalCacheRegistry;
    }
 
    @Start
    protected void start() {
+      Set<Class<?>> indexedEntities = cache.getCacheConfiguration().indexing().indexedEntities();
+      this.indexedEntities = indexedEntities.isEmpty() ? null : indexedEntities.toArray(new Class<?>[indexedEntities.size()]);
+      this.queryKnownClasses = indexedEntities.isEmpty() ? new QueryKnownClasses(cache.getName(), cache.getCacheManager(), internalCacheRegistry) : new QueryKnownClasses(indexedEntities);
+      this.searchFactoryHandler = new SearchFactoryHandler(this.searchFactory, this.queryKnownClasses, new TransactionHelper(transactionManager));
       if (indexedEntities == null) {
```
Comment (anistor) [inline query/src/main/java/org/infinispan/query/backend/QueryInterceptor.java]: Here we should actually refer to ```this.indexedEntities```. This explains the ```AsyncBackendTest``` failures.

**Comment by anistor** [comment]
I'll fix the small error and backport to 9.0.x also.

**Comment by wburns** [comment]
@anistor thanks! :)

**Comment by anistor** [comment]
Fixed on master, and now it's in 9.0.x too: https://github.com/infinispan/infinispan/commit/46c38106de810f52cc861dd7be405df59a7e8ae0

**Pull request merged**